<p>
    2.1 A production manager believes that 100 hours are required to complete a particular task. The accountant, however, disagrees and believes that the task takes less time. To test the view of the accountant, the time taken to complete the task was recorded on 60 occasions by an independent observer. The mean was found to be 96 hours and the standard deviation to be 2.5 hours. Formulate and perform an appropriate test at 5% significance level.
</p>

<p>
    In this example we only have summary statistics so we can't call the R t.test() function. Instead we will calculate the observed test statistic by hand, then call the pt() function to find the p-value. Note however, that we need to be very careful to pass the correct parameters to pt() so that the probability we calculate is appropriate to the test we are performing. We will need to consider whether it is one or two-tailed, and also whether the alternative hypothesis implies an upper or a lower tail. Remember that pt(tobs,df=x) will give the probability of the t distribution on x degrees of freedom lying below the value given by tobs.

In [70]:
# First calculate the observed test statistic. This is a one sample 
# one-tailed (lower tailed) t-test.

# H0: mu=100  H1: mu<100

xbar<-96
mu<-100
n<-60
s<-2.5
dp<-3  # number of decimal places to print

tobs <- (xbar-mu)/(s/sqrt(n))
sprintf("Observed test statistic is %.*f", dp, tobs)

pvalue <- pt(tobs,df=n-1)
sprintf("p-value is %.*f", dp, pvalue)

[1] "Observed test statistic is -12.394"

[1] "p-value is 0.000"

<p>
    As the pvalue is less than 0.05 we reject the null hypothesis and conclude that there is sufficient evidence to suggest that the meantime taken to complete the task is less than 100.
</p>

<p>
    2.2 Twelve packages are randomly selected from the production of breakfast cereal. Their weights in grams are:
</p>
<p>
    288 294 293 294 297 299 301 285 291 298 304 303
</p>
<p>
If production is working correctly, the weights should be independent normally distributed with mean weight of 300 grams.
</p>
<ol type="a">
    <li>Calculate a 95% confidence interval for the mean.
    </li>
    <li>Does this show that the production is working correctly?
    </li>
</ol>

<p>
    Here we have one sample and we want to generate a confidence interval. We have the data so we are going to use the t.test() function. Note that as we only want to pull a confidence interval out of t.test() we do not even need to specify what our hypothesised population mean is. However, we do this anyway by using the mu parameter to t.test()
</p>

In [71]:
# first set up a vector with the data

wts<-c(288,294,293,294,297,299,301,285,291,298,304,303)

# we want to generate a confidence interval for the population mean 
# from a small sample of data. So, we will use t.test() stating a 
# two-sided test to get the confidence interval.

t.test(wts,alternative="two.sided",mu=300)


	One Sample t-test

data:  wts
t = -2.6137, df = 11, p-value = 0.0241
alternative hypothesis: true mean is not equal to 300
95 percent confidence interval:
 291.8641 299.3026
sample estimates:
mean of x 
 295.5833 


<p>
    The 95% confidence interval for the true population mean is (291.8,299.3) so we can see that 300g is not a plausible value of the confidence interval and therefore is not a plausible value for the true population mean weight. (Note how this is confirmed by the result of the hypothesis test where the p-value is 0.0241.)
</p> 

<p>
    2.3 A dispute exists between workers on two production lines. The workers on production line A claim that they are paid less than those on production line B. The company investigates the claim by examining the pay of 70 workers from each production line. The results were as follows:
<table>
<thead>
  <tr>
    <th>Sample Statistics</th>
    <th>Line A</th>
    <th>Line B</th>
  </tr>
</thead>
<tbody>
  <tr>
    <td>mean</td>
    <td>£193.00</td>
    <td>£194.50</td>
  </tr>
  <tr>
    <td>standard deviation</td>
    <td>£6.00</td>
    <td>£7.50</td>
  </tr>
</tbody>
</table>
<p>
    Formulate and perform an appropriate test.
</p>

<p>
    Again we only have the summary statistics for this example. Now, there is a function tsum.test() which you can find in the PASWR package. However, I have had problems loading packages over the last week, and a general problem with my R installation so I am going to calculate the test statistic by hand. However, I encourage you to look up this function and try to use it. This will help you broaden your experience using R.
</p>

In [72]:
# First calculate the observed test statistic. This is two independent  
# groups test. one-tailed (upper)
# H0: mu1=mu2  H1: mu1<mu2 ie. difference <0

xbar1<-193
s1<-6
n1<-70
xbar2<-194.5
s2<-7.5
n2<-70
dp<-3  # number of decimal places to print

# pooled variance
sp2 = ((n1-1)*s1^2+(n2-1)*s2^2)/(n1+n2-2)
tobs <- (xbar1-xbar2)/(sqrt(sp2*(1/n1+1/n2)))

sprintf("Observed test statistic is %.*f", dp, tobs)

pvalue <- pt(tobs,df=n1+n2-2)
sprintf("p-value is %.*f", dp, pvalue)

[1] "Observed test statistic is -1.307"

[1] "p-value is 0.097"

<p>
    As the pvalue is >0.05 we cannot reject the null hypothesis. We conclude that there is insufficient evidence to suggest that the mean pay for production line A is less than production line B.
</p>

<p>
    2.4 A company claims that women take more time off through illness than men. The union wish to test this claim and are able to select two random samples from the workforce: a sample of 17 women and a sample of 11 men. From the personnel record it is found that for the sample of women, the mean number of days illness is 6.3 with a variance of 5.8. For the mean the men is 5.1 days with a variance of 2.6. To help the union in arguing its case, formulate and perform an appropriate test.
</p>

<p>
    Again we only have summary statistics, and again a one-tailed test. I am going to work out the test statistic by hand and calculate the p-value.
</p>

In [73]:
# First calculate the observed test statistic. This is two independent  
# groups test. one-tailed (upper)
# H0: mu1=mu2  H1: mu1>mu2 ie. difference <0 where group 1 is the group
# of women.

# NOTE we have been given sample variances not standard deviations in 
# this question. I won't take the square root as we do want the
# sample variances when we calculate the pooled variance.

xbar1<-6.3
s12<-5.8
n1<-17
xbar2<-5.1
s22<-2.6
n2<-11
dp<-3  # number of decimal places to print

# pooled variance
sp2 = ((n1-1)*s12+(n2-1)*s22)/(n1+n2-2)
tobs <- (xbar1-xbar2)/(sqrt(sp2*(1/n1+1/n2)))

sprintf("Observed test statistic is %.*f", dp, tobs)
# Now we have to be very careful when calling pt() here, as this is an
# upper one-tailed test. By default pt() returns the probability of the 
# a random variable selected from the t-distribution being less than 
# tobs .This works for a lower one-tailed test. We use the parameter 
# lower.tail set to FALSE to indicate that we are seeking the
# probability of lying in the upper tail. Note by default this parameter
# is TRUE.
pvalue <- pt(tobs,df=n1+n2-2, lower.tail=FALSE)
sprintf("p-value is %.*f", dp, pvalue)

[1] "Observed test statistic is 1.451"

[1] "p-value is 0.079"

<p>
    As the pvalue is >0.05 we cannot reject the null hypothesis. We conclude that there is insufficient evidence to suggest that the mean number of sick days for women is greater than that for men.
</p>

<p>2.5 The time taken to complete a particular task was recorded for a sample of eight employees before and after a period of training is shown below (times shown in minutes). Perform a test to investigate whether the training has been effective.
<table>
<thead>
  <tr>
    <th>Employee</th>
    <th>Before Training</th>
    <th>After Training</th>
  </tr>
</thead>
<tbody>
  <tr>
    <td>1</td>
    <td>15</td>
    <td>13</td>
  </tr>
  <tr>
    <td>1</td>
    <td>14</td>
    <td>15</td>
  </tr>
  <tr>
    <td>1</td>
    <td>18</td>
    <td>15</td>
  </tr>
  <tr>
    <td>4</td>
    <td>14</td>
    <td>13</td>
  </tr>
  <tr>
    <td>5</td>
    <td>15</td>
    <td>13</td>
  </tr>
  <tr>
    <td>6</td>
    <td>17</td>
    <td>16</td>
  </tr>
  <tr>
    <td>7</td>
    <td>13</td>
    <td>14</td>
  </tr>
  <tr>
    <td>8</td>
    <td>12</td>
    <td>12</td>
  </tr>
</tbody>
</table>

<p>
    Here we actually have the sample data so we are going to call t.test(). This is paired data, and as we want to investigate whether time taken has improved we testing if the time taken after training is less than before.
</p>

In [74]:
# This is paired data so we want to use a paired t-test. It is 
# one-tailed (upper)
# H0: mud=0  H1: mud>0 

# first set up vectors with the data

timeb<-c(15,14,18,14,15,17,13,12)
timea<-c(13,15,15,13,13,16,14,12)

# 

t.test(timeb,timea,alternative="greater", paired=TRUE)


	Paired t-test

data:  timeb and timea
t = 1.6977, df = 7, p-value = 0.06668
alternative hypothesis: true difference in means is greater than 0
95 percent confidence interval:
 -0.1014435        Inf
sample estimates:
mean of the differences 
                  0.875 


<p>
    As the pvalue is >0.05 we cannot reject the null hypothesis. We conclude that there is insufficient evidence to suggest that the mean time taken has improved after the test.
</p>

<p>
    2.6  A report on a study gives the mean amount spent on text books in an academic year of a random sample of 27 1st year finance students as £114.9 and the standard deviation as £9.3.
</p>
<p>
    Does this data show that 1st year finance students spend more than mathematics students who spend on average £111.3?
</p>

<p>
    We have summary statistics here, and note that this is a single sample test. All we know about the mathematics students is the average amount they spend. So, we are performing a test about the finance students.
</p>

In [75]:
# First calculate the observed test statistic. This is a one sample 
# one-tailed (lower tailed) t-test.

# H0: mu=111.3  H1: mu>111.3

xbar<-114.9
mu<-111.3
n<-27
s<-9.3
dp<-3  # number of decimal places to print

tobs <- (xbar-mu)/(s/sqrt(n))
sprintf("Observed test statistic is %.*f", dp, tobs)

# Now we have to be very careful when calling pt() here, as this is an
# upper one-tailed test. By default pt() returns the probability of the 
# a random variable selected from the t-distribution being less than 
# tobs .This works for a lower one-tailed test. We use the parameter 
# lower.tail set to FALSE to indicate that we are seeking the
# probability of lying in the upper tail. Note by default this parameter
# is TRUE.
pvalue <- pt(tobs,df=n-1,lower.tail=FALSE)
sprintf("p-value is %.*f", dp, pvalue)

[1] "Observed test statistic is 2.011"

[1] "p-value is 0.027"

<p>
    As the pvalue is less than 0.05 we reject the null hypothesis and conclude that there is sufficient evidence to suggest that the mean amount spent on books by finance students is greater than the average spent by mathemtatics students.
</p>

<p>
    2.7 A bank compares two proposals to increase the amount that its credit card customerscharge on their cards. Proposal A offers to eliminate the annual fee for customers who charge £2400 or more during the year. Proposal B offers a small percent of the total charged as a cash rebate at  the end of the year. The bank offers the proposals to a total sample of 35 of its existing credit card customers. At the end of the year, the total  amount charged by each customer is recorded. The summary statistics follow:
<table>
<thead>
  <tr>
    <th>Group</th>
    <th>sample mean</th>
    <th>sample standard deviation</th>
  </tr>
</thead>
<tbody>
  <tr>
    <td>A</td>
    <td>£1987</td>
    <td>£392</td>
  </tr>
  <tr>
    <td>B</td>
    <td>£2056</td>
    <td>£413</td>
  </tr>
</tbody>
</table>
<p>
    We want to answer the question: Do the data show a significant difference between the mean amounts charged by customers offered the two plans?
</p>

<p>
    Here we again have summary statistics so we are going to calculate the observed test statistic and then calculate the p-value using pt(). This is a comparison of means from two separate groups so we will use the independent groups t-test. As we are looking for a difference between the mean amounts this is a two-tailed test.
</p>

In [76]:
# First calculate the observed test statistic. This is two independent  
# groups test. two-tailed
# H0: mu1=mu2  H1: mu1 not equal to mu2 ie. difference not equal to 0 

xbar1<-1987
s1<-392
n1<-15
xbar2<-2056
s2<-413
n2<-20
dp<-3  # number of decimal places to print

# pooled variance
sp2 = ((n1-1)*s1^2+(n2-1)*s2^2)/(n1+n2-2)
tobs <- (xbar1-xbar2)/(sqrt(sp2*(1/n1+1/n2)))

sprintf("Observed test statistic is %.*f", dp, tobs)
# Now we have to be very careful when calling pt() here, as this is a
# two-tailed test, and we need to double the probability. We also don't
# know whether our observed test statistic is negative or not.
# Therefore, we take the absolute value of the observed test statistic
# and call pt() as if looking at the upper tail of the distribution.
# Note by default pt() returns the probability of the 
# a random variable selected from the t-distribution being less than 
# tobs.This works for a lower one-tailed test. We use the parameter 
# lower.tail set to FALSE to indicate that we are seeking the
# probability of lying in the upper tail. Note by default this parameter
# is TRUE.
pvalue <- 2*pt(abs(tobs),df=n1+n2-2, lower.tail=FALSE)
sprintf("p-value is %.*f", dp, pvalue)

[1] "Observed test statistic is -0.500"

[1] "p-value is 0.621"

<p>
    As the pvalue is >0.05 we cannot reject the null hypothesis. We conclude that there is insufficient evidence to suggest that the mean amount spent is different between group A and group B.
</p>

<p>
    2.8  A commercial educational software producer advertises a software program designed to improve scores on a verbal test. A random sample of 10 students took the verbal test before and after using this software. The results appear below:
</p>
<table>
<thead>
  <tr>
    <th>Student no.</th><th>1</th><th>2</th><th>3</th><th>4</th>
    <th>5</th><th>6</th><th>7</th><th>8</th><th>9</th><th>10</th>
  </tr>
</thead>
<tbody>
    <tr>
        <td>Score Before</td><td>430</td><td>600</td><td>480</td>
        <td>500</td><td>610</td><td>380</td><td>570</td><td>490</td>
        <td>540</td><td>370</td>
    </tr>
    <tr>
        <td>Score After</td><td>450</td><td>640</td><td>490</td><td>550</td><td>630</td><td>420</td><td>610</td><td>490</td><td>575</td><td>400</td>
    </tr>
</tbody>
</table>

Should we conclude that the software improves the verbal scores?

<p>
    Here we actually have the sample data so we are going to call t.test(). This is paired data, and as we want to investigate whether scores have improved. So taking the difference as before-after this is a lower one-tailed test.
</p>

In [77]:
# This is paired data so we want to use a paired t-test. It is 
# one-tailed (lower)
# H0: mud=0  H1: mud<0  so this is taking the difference as before-after

# first set up vectors with the data

scoreb<-c(430,600,480,500,610,380,570,490,540,370)
scorea<-c(450,640,490,550,630,420,610,490,575,400)

# Now call t.test for paired data. Make sure that the alternative here
# is consistent with the order the difference is taken.

t.test(scoreb,scorea,alternative="less", paired=TRUE)


	Paired t-test

data:  scoreb and scorea
t = -5.7611, df = 9, p-value = 0.0001362
alternative hypothesis: true difference in means is less than 0
95 percent confidence interval:
     -Inf -19.4317
sample estimates:
mean of the differences 
                  -28.5 


<p>
    As the pvalue is less than 0.05 we reject the null hypothesis in favour of the alternative. We conclude that there is sufficient evidence to suggest that the mean test score is greater after using the software.
</p>

<p>
    2.9  The average weekly overtime earnings (in £s) from a sample of workers this year and last year were recorded as follows:
<table>
<thead>
  <tr>
    <th>Sample statistics</th>
    <th>last year</th>
    <th>this year</th>
  </tr>
</thead>
<tbody>
  <tr>
    <td>number</td>
    <td>28</td>
    <td>25</td>
  </tr>
  <tr>
    <td>mean</td>
    <td>3.00</td>
    <td>3.50</td>
  </tr>
  <tr>
    <td>standard deviation</td>
    <td>2.98</td>
    <td>3.06</td>
  </tr>
</tbody>
</table>
<p>
<ol type="a">
    <li>Construct a 95% confidence interval for the increase in average weekly overtime earnings.
    </li>
    <li>Is there a ‘statistically significant’ difference in the average earnings between this year and last year? (Interpret the confidence interval in (1) but also perform an appropriate test.)
    </li>
</ol>
</p>

<p>
    These are summary statistics so we are going to have to calculate the confidence interval by hand. The sample is for two independent groups.
</p>

In [78]:
# First calculate the observed test statistic. This is two independent  
# groups test. 

xbar1<-3.5
s1<-3.06
n1<-25
xbar2<-3.0
s2<-2.98
n2<-28
dp<-3  # number of decimal places to print

# pooled variance
sp2 = ((n1-1)*s1^2+(n2-1)*s2^2)/(n1+n2-2)

# now find the lower and upper bounds of the 95% confidence interval
# for the difference in means.
# first we find the critical value for 2.5% of the t distribution lying
# above. To do this we use qt() which gives quantiles for the t
# distribution on given number of degrees of freedom. Note that as we
# want a positive value we look for probability 0.975
tcrit <- qt(0.975,n1+n2-2)
cilow <- xbar1-xbar2-tcrit*sqrt(sp2*(1/n1+1/n2))
cihigh <- xbar1-xbar2+tcrit*sqrt(sp2*(1/n1+1/n2))
                                 
sprintf("95%% confidence interval is (%.*f,%.*f) ",dp,cilow,dp,cihigh)

tobs <- (xbar1-xbar2)/(sqrt(sp2*(1/n1+1/n2)))


[1] "95% confidence interval is (-1.167,2.167) "

<p>
    We are 95% confident that the true mean difference lies within the confidence interval. A mean difference of 0 is a plausible value of the confidence interval.
</p>

In [79]:
# First calculate the observed test statistic. This is two independent  
# groups test.
# H0: mu1=mu2  H1: mu1 not equal to mu2 ie. difference not equal to 0 

xbar1<-3.5
s1<-3.06
n1<-25
xbar2<-3.0
s2<-2.98
n2<-28
dp<-3  # number of decimal places to print

# pooled variance
sp2 = ((n1-1)*s1^2+(n2-1)*s2^2)/(n1+n2-2)

tobs <- (xbar1-xbar2)/(sqrt(sp2*(1/n1+1/n2)))

sprintf("Observed test statistic is %.*f", dp, tobs)
# Now we have to be very careful when calling pt() here, as this is a
# two-tailed test, and we need to double the probability. We also don't
# know whether our observed test statistic is negative or not.
# Therefore, we take the absolute value of the observed test statistic
# and call pt() as if looking at the upper tail of the distribution.
# Note by default pt() returns the probability of the 
# a random variable selected from the t-distribution being less than 
# tobs.This works for a lower one-tailed test. We use the parameter 
# lower.tail set to FALSE to indicate that we are seeking the
# probability of lying in the upper tail. Note by default this parameter
# is TRUE.
pvalue <- 2*pt(abs(tobs),df=n1+n2-2, lower.tail=FALSE)
sprintf("p-value is %.*f", dp, pvalue)

[1] "Observed test statistic is 0.602"

[1] "p-value is 0.550"

<p>
    As the pvalue is >0.05 we cannot reject the null hypothesis. We conclude that there is insufficient evidence to suggest that the mean earnings is different between the two groups from different years.
</p>

In [80]:
# Example 3.1 I am going to build the data frame in two ways just to show
# you different ways of setting the data up. Use this to check your
# hand-written solutions.

# First just build the data frame by hand

ans.df<-data.frame(child=seq(1,24,1),
                   correct=c(9,8,8,9,7,7,
                             2,5,4,3,4,3,
                             9,3,7,8,5,6,
                             5,7,3,6,7,4),
                   group=c(1,1,1,1,1,1,
                           2,2,2,2,2,2,
                           3,3,3,3,3,3,
                           4,4,4,4,4,4))

with (ans.df, {
  groups<-factor(group)
  aov1<-aov(correct~groups)
  summary(aov1)
})

# The Observed Test Statistic 9.212 is in the Rejection Region (it 
# is > 3.10) and the p-value is less than 0.001 so we can reject H0 in
# favour of H1 at the 5% level. This there is sufficient evidence that
# the mean number of correct answers is not the same for all four pattern
# recognition skills groups.

# Now going to read the data in from the spreadsheet

library("readxl")

Lab3data <- read_excel("MATPMD1LabSession3Data.xlsx")

with (Lab3data,{
  groups<-factor(Q1Group)
  aov1<-aov(Q1Correct~groups)
  
  summary(aov1)
})


            Df Sum Sq Mean Sq F value   Pr(>F)    
groups       3  63.79  21.264   9.212 0.000495 ***
Residuals   20  46.17   2.308                     
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

            Df Sum Sq Mean Sq F value   Pr(>F)    
groups       3  63.79  21.264   9.212 0.000495 ***
Residuals   20  46.17   2.308                     
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
3 observations deleted due to missingness

In [81]:
# Example 3.2 Use this to check your hand calculations

library("readxl")

Lab3data <- read_excel("MATPMD1LabSession3Data.xlsx")
with (Lab3data,{

  strain<-factor(Q2Strain)
  aov2<-aov(Q2Days~strain)
  summary(aov2)

})

# The Observed Test Statistic (12.65) is in the Rejection Region (>3.40) and
# p-value is < 0.0001 (NB P(F(2.24)>9.34)=0.0001) so we can reject H0 in
# favour of H1 at the 5% level and conclude that there is sufficient
# evidence at the 5% level fo a difference somewhere in the mean number of
# days to death for the three strains of mice.

            Df Sum Sq Mean Sq F value   Pr(>F)    
strain       2  87.63   43.81   12.65 0.000177 ***
Residuals   24  83.11    3.46                     
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

In [82]:
# Example 3.3

library("readxl")

Lab3data <- read_excel("MATPMD1LabSession3Data.xlsx")

  with(Lab3data,{
  
  groups<-factor(Q3Group)
  aov3<-aov(Q3BPFall~groups)
  summary(aov3)

})
  
# The Observed Test Statistic (10.15) is in the Rejection Region (greater
# than 3.49) and p-value is < 0.001 (Since P(F(3,12)>10.80=0.001) so we can
# reject H0 in favour of H1 at the 5% leveland conclude there is sufficient
# evidence that the mean falls in SBP are not all the same.

            Df Sum Sq Mean Sq F value Pr(>F)   
groups       3  57.69  19.229   10.15 0.0013 **
Residuals   12  22.73   1.894                  
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
11 observations deleted due to missingness

In [83]:
# 3.4

# set up contingency table for males
(maledf <- matrix(c(10,10,20,4,9,11),ncol=3))

# run chi-squared test for association for males
chisq.test(maledf)


[,1] [,2] [,3]
[1,] 10   20    9  
[2,] 10    4   11


	Pearson's Chi-squared test

data:  maledf
X-squared = 8.1964, df = 2, p-value = 0.0166


In [84]:
# set up contingency table for females
(femaledf <- matrix(c(18,12,18,4,12,8),ncol=3))

# run chi-squared test for association for males
chisq.test(femaledf)

[,1] [,2] [,3]
[1,] 18   18   12  
[2,] 12    4    8


	Pearson's Chi-squared test

data:  femaledf
X-squared = 3.2727, df = 2, p-value = 0.1947


In [85]:
# set up contingency table for both males and females
(bothdf <- matrix(c(28,22,38,8,21,19),ncol=3))

# run chi-squared test for association for males
chisq.test(bothdf)

[,1] [,2] [,3]
[1,] 28   38   21  
[2,] 22    8   19


	Pearson's Chi-squared test

data:  bothdf
X-squared = 10.595, df = 2, p-value = 0.005005


In [86]:
# 3.5 goodness of fit for uniform distribution with mean 45/5=9. On
# call to chisq.test() there is equal probability of rejection on
# each day.

reject<-c(15,8,5,5,12)
probs<-c(1/5,1/5,1/5,1/5,1/5)

chisq.test(reject,p=probs)


	Chi-squared test for given probabilities

data:  reject
X-squared = 8.6667, df = 4, p-value = 0.06999
